### ISSUE TITLE
Return Detailed Tracking Information For Tracker Emblems

### ISSUE THREAD

**Issue by lowlines** [Bungie-net/api#482]
While it is great having emblems that tell you how many of x you have found at a given destination, or the new Data Recovered stat on the Hellas Basin map, it becomes difficult to keep track of what users have and don't have when there is no in-game way to see which ones they are missing.

Much like Calcified fragments were returned in the D1 APIs, it would be awesome if there was a way to return objects like region chests, lost sectors, sleeper nodes, data fragments and future destination collectibles as a series of flags.

I've been working on a web version of the in-game maps and adding markers not shown in game and while I am planning on adding a way to manually check off things, it'd be better that I could check a user's account and update that list dynamically.

https://lowlidev.com.au/destiny/maps/mars

**Comment by vthornheart-bng** [comment]
I passed the info on, I think it'll be good for our designers to know that this is a feature that people would find useful!  Have those of you making apps/sites so far found that people have been wishing this existed in the data?  I know Calcified Fragments were popular in D1 so that serves as a good precedent, but feel free to add any info you have about users requesting this for D2 here!

**Comment by mattybeard** [comment]
I absolutely think tracking of specific items in a check list form is an amazing thing for both developers and end users - hunting ghosts in D1 was amazing and the sites used to find them were special and a bit of me was sad when they weren't in D2.

Even looking at Grimoire, I know the Grimoire section of DestinyStatus and similar that showed what grimoire you had, what count towards it and stuff kept people coming back for more and more.

**Comment by lowlines** [comment]
I'm seeing threads like this one surrounding tracked objects like the sleeper nodes.
https://twitter.com/ItsDJonTwitch/status/996428171808575489

**Comment by lowlines** [comment]
I get a lot of engagement in tweets I post about tracking stuff on maps. Folks will also often ask or expect the APIs to return this info because it was something they did in D1.
https://twitter.com/lowlines/status/996429398487150592

**Comment by designker** [comment]
Yes it was great in D1 to be able to create sites or apps that knew which specific scanable/dead ghost/fragment was missing so we could link to a specific guide about how to find them.

**Comment by vthornheart-bng** [comment]
All great info, I appreciate it!  Because there's so few of us working on the API (in the end, it's just Achronos and I these days) and there's so many non-API features we have to also work on, it's often a struggle to find what's *really* worth doing with the short time we have.  This kind of information is extremely valuable for helping us figure that out.  I appreciate it!

**Comment by Barbarian86** [comment]
Adding tracking information for Sleeper Nodes would be an amazing place to start, as these don’t disappear once we’ve activated/opened them the way data caches do after we destroy them. I’m absolutely loving the Warmind expansion overall, but getting duplicate after duplicate and not knowing for certain if it’s one I’ve already collected is a bit discouraging.

I’m not sure how simple of a fix it would be on your end, but any effort or feedback on this is greatly appreciated. Thank you (and Achronos) for your time.

**Comment by lowlines** [comment]
My Hellas Basin map kind of blew up over the last couple of days. I've also gotten a ton of requests for a way to check which Data Fragments and Sleeper Nodes folks already have. Just look through the comments!

https://twitter.com/lowlines/status/997816426198020096
https://www.reddit.com/r/DestinyTheGame/comments/8kiafu/an_interactive_map_of_hellas_basin_mars/

**Comment by vthornheart-bng** [comment]
Ah, yes - we're working on getting some more tracking information going!  I'll try to get you all some more information tomorrow.  There's a ton of things we could and would like to track, but it'll probably end up coming in waves based on what we need right away for an upcoming feature (that gave us the time/resources to finally start working on this!) vs. what would be "nice to have."  I'll keep you all posted as best I can.

**Comment by michabbb** [comment]
@vthornheart-bng because you say it´s important for you to know what people are looking for , maybe it´s worth using something like [uservoice](https://www.uservoice.com/) ?

**Comment by designker** [comment]
We can already vote here. Pointing out the kind of useful service you would build with a specific piece of info seems to be the best way to raise up what would be useful.

**Comment by vthornheart-bng** [comment]
Okay, so to loop back to this... the following will be checklists that will come out soon - hopefully by our next deployment:

* Adventures Completed (per character)
* Region Chests obtained (account wide)
* Latent Memories obtained (account wide)
* Sleeper nodes (account wide)

Some that I'm intending to grab but haven't had time to wire up yet:

* Cayde Journals obtained (account wide)
* Lost Sectors found (per character)
* Ghost Lore found (account wide)

Note that we will only have additional information for some of these.  For example, we don't have location information for Sleeper Nodes or Latent Memories, which means those wanting to build location-based features for them will unfortunately have to have their own manually constructed data source.  But where possible, we will return whatever associations we were able to make (for instance, we'll associate the sleeper nodes "checklist" items with the related items you receive in order to find them).

**Comment by philkernick** [comment]
Region chests are per-character, not per-account on Mercury and Mars.  Will it return a chest as completed if it has been found on any character, or only once it has been found on all characters?

**Comment by vthornheart-bng** [comment]
Okay, so I am utterly swamped with work: I won't have time to write up a full guide... but hopefully the documentation released tomorrow with the deployment will be enough to help get you folks going with this data in the interim.

(also, indeed you are correct - Region Chests are per-character.  For anything that is per character, we'll return it in the characterProgressions component!  Account-wide ones will be returned in the new profileProgressions component!)

One interesting caveat is about Adventures.  It's a known limitation of the data we're storing about whether you've completed adventures, but *most* of our progress is stored on a per-account level for adventure completions.  However, DLC1 and DLC2 adventures were stored on a per-character level.  This will more than likely not be fixed in the future, so note that we'll return Adventures per-character but indeed any pre-DLC Adventures will be able to be "checked off" by any individual character.  Not ideal, but we work with what we've got. :)

I managed to grab the following.  Note that the asterisk means that this data exists, but is in "beta": please let us know if you run into any problems with them!

Adventures (Per character)
Region Chests (Per character)
Latent Memories
Raid Lairs
Ghost Lore*
Cayde Journals*
Lost Sectors* (Per character)
Sleeper Nodes*

**Comment by vthornheart-bng** [comment]
Correction - I keep saying Adventures when I mean Region Chests.

Region Chests are the ones with mixed account and character statuses.  Adventures are purely character-level.  So Region Chests are the ones with the issue mentioned above!  Thanks @floatingatoll for noticing that. :)
